haystack_experimental/components/tools/openapi/_openapi.py

```diff
@@ -169,11 +170,32 @@ def get_auth_function(self) -> Callable[[Dict[str, Any], Dict[str, Any]], Any]:
             )
         raise ValueError(f"Unsupported credentials type: {type(self.credentials)}")
 
-    def get_tools_definitions(self) -> List[Dict[str, Any]]:
+    def get_tools_definitions(
+            self,
+            normalization_fn: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None
```

Comment:
I don't see any callsites where this parameter is actually used? I'd remove it and use the default normalization function. All of this is internal, so it can be reintroduced if we end up needing the extra flexibility.

Besides, given how most LLM providers take after OpenAI w.r.t to API constraints, we can probably get away with just one normalization function.